test/server/admin/admin_test.cc

```diff
@@ -217,6 +222,242 @@ TEST_P(AdminInstanceTest, ConfigDumpMaintainsOrder) {
   }
 }
 
+// Test that using ?includeEds parameter adds EDS to the config dump.
+TEST_P(AdminInstanceTest, ConfigDumpWithEndpoint) {
+  Upstream::ClusterManager::ClusterInfoMap cluster_map;
+  ON_CALL(server_.cluster_manager_, clusters()).WillByDefault(ReturnPointee(&cluster_map));
+
+  NiceMock<Upstream::MockClusterMockPrioritySet> cluster;
+  cluster_map.emplace(cluster.info_->name_, cluster);
+
+  ON_CALL(*cluster.info_, addedViaApi()).WillByDefault(Return(false));
+
+  Upstream::MockHostSet* host_set = cluster.priority_set_.getMockHostSet(0);
+  auto host = std::make_shared<NiceMock<Upstream::MockHost>>();
+
+  envoy::config::core::v3::Locality locality;
+  ON_CALL(*host, locality()).WillByDefault(ReturnRef(locality));
+
+  host_set->hosts_.emplace_back(host);
+  Network::Address::InstanceConstSharedPtr address =
+      Network::Utility::resolveUrl("tcp://1.2.3.4:80");
+  ON_CALL(*host, address()).WillByDefault(Return(address));
+  const std::string hostname = "foo.com";
+  ON_CALL(*host, hostname()).WillByDefault(ReturnRef(hostname));
+
+  auto metadata = std::make_shared<envoy::config::core::v3::Metadata>();
+  ON_CALL(*host, metadata()).WillByDefault(Return(metadata));
+
+  const std::string hostname_for_health_checks = "test_hostname_healthcheck";
+  ON_CALL(*host, hostnameForHealthChecks()).WillByDefault(ReturnRef(hostname_for_health_checks));
+  Network::Address::InstanceConstSharedPtr healthcheck_address =
+      Network::Utility::resolveUrl("tcp://1.2.3.5:90");
+  ON_CALL(*host, healthCheckAddress()).WillByDefault(Return(healthcheck_address));
+
+  ON_CALL(*host, health()).WillByDefault(Return(Upstream::Host::Health::Healthy));
+
+  ON_CALL(*host, weight()).WillByDefault(Return(5));
+  ON_CALL(*host, priority()).WillByDefault(Return(6));
+
+  Buffer::OwnedImpl response;
+  Http::TestResponseHeaderMapImpl header_map;
+  EXPECT_EQ(Http::Code::OK, getCallback("/config_dump?includeEds", header_map, response));
+  std::string output = response.toString();
+  const std::string expected_json = R"EOF({
+ "configs": [
```

Comment:
Please use YAML for literal config examples, it's much neater and that's what the rest of the Envoy tests do.

Reply:
Do you mean using YAML for setup? Or using it as expected output?

Reply:
Because I have found that multiple other test files use YAML for setup or as expected output to verify. However, all the test functions in admin_test use JSON as both input and output. So I am wondering if using YAML follows the origin style of admin_test, could you please clarify it?